src/main/java/com/project/adminmns/controller/ConnexionController.java

```diff
@@ -48,15 +43,55 @@ public ResponseEntity<Map<String, Object>> connexion (@RequestBody ModelUser use
         }
     }
 
+    public static boolean patternMatches(String emailAddress, String regexPattern) {
+        return Pattern.compile(regexPattern)
+                .matcher(emailAddress)
+                .matches();
+    }
 
-    @PostMapping("/inscription")
-    public ResponseEntity<Map<String, Object>> inscription (@RequestBody ModelUser user){
 
-        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+    @PostMapping("/inscription")
+    public ResponseEntity<Map<String, Object>> inscription (@Valid @RequestBody ModelUser user){
+
+        //The following restrictions are imposed in the email address’ local part by using this regex:
+        //
+        //It allows numeric values from 0 to 9.
+        //Both uppercase and lowercase letters from a to z are allowed.
+        //Allowed are underscore “_”, hyphen “-“, and dot “.”
+        //Dot isn’t allowed at the start and end of the local part.
+        //Consecutive dots aren’t allowed.
+
+        //For the local part, a maximum of 64 characters are allowed.
+
+        //Restrictions for the domain part in this regular expression include:
+        //
+        //It allows numeric values from 0 to 9.
+        //We allow both uppercase and lowercase letters from a to z.
+        //Hyphen “-” and dot “.” aren’t allowed at the start and end of the domain part.
+        //No consecutive dots
+        String emailRegexPattern = "^(?=.{1,64}@)[A-Za-z0-9_-]+(\\.[A-Za-z0-9_-]+)*@"
+                + "[^-][A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z]{2,})$";
+
+        //Minimum eight characters,
+        // at least one uppercase letter,
+        // one lowercase letter,
+        // one number
+        // and one special character
+        String passwordRegexPattern = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{8,}$";
+
+        String emailAddress = user.getEmail();
+        String password = user.getPassword();
+
+
+        if (patternMatches(emailAddress, emailRegexPattern) && patternMatches(password, passwordRegexPattern)){
+            user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+            userDao.save(user);
```

Comment:
this.userDAO

Reply:
Tu préfères que je mettent des this. partout ? parce que là ca sert pas à grand chose si ?

Reply:
il me semble que j'en mets partout du coup je sais pas, je trouve ça plus propre